src/main/groovy/com/bmuschko/gradle/docker/tasks/AbstractDockerRemoteApiTask.groovy

```diff
@@ -157,8 +162,9 @@ abstract class AbstractDockerRemoteApiTask extends DefaultTask {
 
         DefaultDockerClientConfig dockerClientConfig = dockerClientConfigBuilder.build()
 
-        // Create client
-        DockerClient dockerClient = DockerClientBuilder.getInstance(dockerClientConfig).build()
+        boolean useNettyExecFactory = Boolean.valueOf(System.getenv(ENV_VAR_GDP_CONN_NETTY))
+        DockerCmdExecFactory execFactory = useNettyExecFactory ? new NettyDockerCmdExecFactory() : null
```

Comment:
I am wondering why we need to switch the `DockerCmdExecFactory` completely instead of configuring the the Jersey implementation. Apparently, the `JerseyDockerCmdExecFactory` doesn't set an unlimited read and connect timeout. Otherwise you wouldn't see the issue with current implementation. I looked around in the code a bit but couldn't find where it does change the defaults. Maybe it's even a somewhat unrelated bug with the `JerseyDockerCmdExecFactory` in general.

Popping in `NettyDockerCmdExecFactory` just so an unlimited timeout is set (which seems to be the default for it) sounds a little backwards. Moreover, I'd personally consider switching out the communication layer pretty high risk (especially because it's not the default in Docker Java right). I don't think we can be sure that the Netty implementation just works the same way as the Jersey layer without hick ups.

Reply:
@bmuschko I think we're getting caught up on the `timeout` property so let me stop mentioning it. It's the the implementation in general that seems to be causing things to pass ... for whatever reason. I honestly don't know. When I run the tests locally with this change, and our pipeline where things do fail, things pass. The process itself takes maybe 10 minutes at most and doesn't trip any timeout anyway. Again ... I don't know what to make of it other than it just works.

Reply:
FWIW: following the code for either jersey or netty impl the timeouts, if not set, will default to infinity.

Reply:
I think I caught up by the example you gave initially. I didn't quite understand what you meant by "the timeout is just an example". It's interesting that Docker Java still didn't address this crucial error after 3 years.

BTW: Did you see this comment https://github.com/docker-java/docker-java/issues/698#issuecomment-403502833 that could work around the issue without have to switch to Netty? Maybe that's an easier, less invasive solution to the problem.

Reply:
That's an interesting solution... let me give it a go and see how things fare. I would still like to get this in though.